src/compas/data/validators.py

```diff
@@ -40,6 +50,21 @@ def validate_data(data, cls):
     ------
     SchemaError
     """
-    import jsonschema
-    jsonschema.validate(data, schema=cls.JSONSCHEMA)
-    return cls.DATASCHEMA.validate(data)
+    from jsonschema import RefResolver, Draft7Validator
+
+    here = os.path.dirname(__file__)
+
+    schema_name = '{}.json'.format(cls.__name__.lower())
+    schema_path = os.path.join(here, 'schemas', schema_name)
+    with open(schema_path, 'r') as fp:
+        schema = json.load(fp)
+
+    definitions_path = os.path.join(here, 'schemas', 'compas.json')
+    with open(definitions_path, 'r') as fp:
+        definitions = json.load(fp)
+
+    resolver = RefResolver.from_schema(definitions)
+    validator = Draft7Validator(schema, resolver=resolver)
+    validator.validate(data)
+
+    return json.loads(json.dumps(data, cls=DataEncoder), cls=DataDecoder)
```

Comment:
The whole point of the validation is so that this loads/dumps combo will raise some exception if the `data` isn't valid, right?  Maybe it would be nice to wrap this in a `try/except` statement which would basically return the error that `json` raises, but wrapped in something that says "Sorry, compas validation failed"